src/lib/Zikula/Common/Translator/Translator.php

```diff
@@ -0,0 +1,262 @@
+		$this->loaderIds = $loaderIds;
+		// check option names
+		if ($diff = array_diff ( array_keys ( $options ), array_keys ( $this->options ) )) {
+			throw new \InvalidArgumentException ( sprintf ( 'The Translator does not support the following options: \'%s\'.', implode ( '\', \'', $diff ) ) );
+		}
+		
+		$this->domain = 'zikula';
+		
+		$this->options = array_merge ( $this->options, $options );
+		$this->resourceLocales = array_keys ( $this->options ['resource_files'] );
+		if (null !== $this->options ['cache_dir'] && $this->options ['debug']) {
+			$this->loadResources ();
+		}
+		parent::__construct ( $container->getParameter ( 'kernel.default_locale' ), $selector, $this->options ['cache_dir'], $this->options ['debug'] );
+	}
+	
+    /**
+     * {@inheritdoc}
+     */
+	public function warmUp($cacheDir) {
+		// skip warmUp when translator doesn't use cache
+		if (null === $this->options ['cache_dir']) {
+			return;
+		}
+		foreach ( $this->resourceLocales as $locale ) {
+			$this->loadCatalogue ( $locale );
+		}
+	}
+	
+    /**
+     * {@inheritdoc}
+     */
+	protected function initializeCatalogue($locale) {
+		$this->initialize ();
+		parent::initializeCatalogue ( $locale );
+	}
+	
+	/**
+	 * {@inheritdoc}
+	 */	
+	protected function initialize() {
+		$this->loadResources ();
+		foreach ( $this->loaderIds as $id => $aliases ) {
+			foreach ( $aliases as $alias ) {
+				$this->addLoader ( $alias, $this->container->get ( $id ) );
+			}
+		}
+	}
+	
+	/**
+	 * {@inheritdoc}
+	 * @todo better load resource
+	 */
+	private function loadResources() {
+		foreach ( $this->options ['resource_files'] as $locale => $files ) {
+			foreach ( $files as $key => $file ) {
+				
+				$c = substr_count ( $file, "." );
+				
+				if ($c < 2) {
+					// explode('.', basename($file), 3);
+					// filename is domain.locale.format
+					list ( $domain, $format ) = explode ( '.', basename ( $file ), 2 );
+				} else {
+					// explode('.', basename($file), 3);
+					// filename is domain.locale.format
+					list ( $domain, $locale, $format ) = explode ( '.', basename ( $file ), 3 );
+				}
+				
+				list ( $domain, $format ) = explode ( '.', basename ( $file ), 2 );
+				
+				$this->addResource ( $format, $file, $locale, $domain );
+				unset ( $this->options ['resource_files'] [$locale] [$key] );
+			}
+		}
+	}
+	
+	/**
+	 * Set the translation domain.
+	 *
+	 * @param string $domain
+	 *        	Gettext domain.
+	 *        	
+	 * @return void
+	 */
+	public function setDomain($domain = null) {
+		$this->domain = $domain;
+	}
+	
+	/**
+	 * Get translation domain.
+	 *
+	 * @return string $this->domain
+	 */
+	public function getDomain() {
+		return $this->domain;
+	}
+	
+    /**
+     * {@inheritdoc}
+     */
+	public function trans($id, array $parameters = array(), $domain = null, $locale = null) {
+		if (null === $domain) {
+			$domain = $this->domain;
+		}
+		
+		return strtr ( $this->getCatalogue ( $locale )->get ( ( string ) $id, $domain ), $parameters );
```

Comment:
CS again (and else where) remove the spaces, e.g.

`return strtr($this->getCatalogue( $locale )->get((string) $id, $domain), $parameters);`